packages/labs/testing/src/lib/fixtures/csr-fixture.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {render} from 'lit';
+import type {LitElement, TemplateResult} from 'lit';
+import type {FixtureOptions} from './fixture-options.js';
+
+/**
+ * Renders the provided Lit template client-side.
+ */
+export async function csrFixture<T extends LitElement>(
+  template: TemplateResult,
+  {modules, base}: FixtureOptions
+): Promise<T> {
+  if (base === undefined) {
+    // Find the test file url from the call stack
+    // Chrome:
+    // Error
+    // at ssrFixture (http://localhost:8000/node_modules/@lit-labs/testing/lib/fixtures/ssrFixture.js:29:17)
+    // at ssrHydratedFixture (http://localhost:8000/node_modules/@lit-labs/testing/lib/fixtures/ssrFixture.js:76:12)
+    // at o.<anonymous> (http://localhost:8000/test/my-element_test.js?wtr-session-id=GhB4vW1TXWxXwqgt3F8QD:65:30)
+    //
+    // Firefox:
+    // ssrFixture@http://localhost:8000/node_modules/@lit-labs/testing/lib/fixtures/ssrFixture.js:29:17
+    // ssrHydratedFixture@http://localhost:8000/node_modules/@lit-labs/testing/lib/fixtures/ssrFixture.js:76:12
+    // @http://localhost:8000/test/my-element_test.js?wtr-session-id=NaCljZAFiyeOoV6-qBqwr:65:30
+    //
+    // Webkit:
+    // @http://localhost:8000/node_modules/@lit-labs/testing/lib/fixtures/ssrFixture.js:29:26
+    // asyncFunctionResume@[native code]
+    // asyncFunctionResume@[native code]
+    // @http://localhost:8000/test/my-element_test.js?wtr-session-id=aKWON-wBOBGyzb2CwIvmK:65:37
+    const match = new Error().stack?.match(
+      /http:\/\/localhost.+(?=\?wtr-session-id)/
+    );
+    if (!match) {
+      throw new Error('Could not find call site for csrFixture');
```

Comment:
In what cases do we expect this to fail? It's unclear in the readme if users are expected to pass `import.meta.url` or if they can reasonably expect this to work without having to pass that. Maybe clarify?

Reply:
I don't anticipate us to ever reach this unless how the stack is generated gets changed (or the regex somehow fails for some browser besides chrome, firefox, webkit) or if web test runner changes the query to something other than `wtr-session-id`.

I've updated the readme to expand more on the `base` option. Leaving it empty should have the same behavior as passing in `import.meta.url`.